src/index.ml

```diff
@@ -420,40 +420,33 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
           l "[%s] no changes detected" (Filename.basename t.root))
     in
     let add_log_entry log e = Tbl.replace log.mem e.key e.value in
+    let sync_log_async ?(generation_change = false) () =
+      match t.log_async with
+      | None -> t.log_async <- try_load_log t (log_async_path t.root)
+      | Some log ->
+          let offset = IO.offset log.io in
+          let new_offset = IO.force_offset log.io in
+          if generation_change || offset <> new_offset then (
+            Tbl.clear log.mem;
+            iter_io (add_log_entry log) ~max:new_offset log.io )
```

Comment:
There should be no need to specify `max` here, `iter_io` will use `IO.offset` anyway, which will return `new_offset` (it's fine to leave it though).